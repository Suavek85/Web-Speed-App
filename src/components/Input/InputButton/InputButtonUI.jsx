import React from 'react'
import PropTypes from 'prop-types'
import cls from 'classnames'

//INTERNAL
import { appStates } from '../../../constants/states'
import { btnsContent } from '../../../constants/content'
import './InputButtonUI.scss'  

function InputButtonUI(props) {
 
  const { getAppState, handleGetReportClick } = props

  //BUTTON FORMATTING - CLASSES AND CONTENT
  const isStatus = state => getAppState === state
  
  const getContentFromStatus = state => { 
    const contentIndex = Object.keys(btnsContent).findIndex(el => el === state)
    return Object.values(btnsContent)[contentIndex]
  }

  const getButtonClasses = cls({
    'inputbutton__wrapper': true,
    'inputbutton__wrapper--inactive': isStatus(appStates.INACTIVE),
    'inputbutton__wrapper--loading': isStatus(appStates.LOADING),
    'inputbutton__wrapper--error': isStatus(appStates.ERROR),
    'inputbutton__wrapper--success': isStatus(appStates.SUCCESS),
  })

  return (
    <>
      <button 
        onClick={handleGetReportClick} 
        disabled={isStatus(appStates.LOADING)}
        className={getButtonClasses}
      >   
        {getContentFromStatus(getAppState)}
      </button>
    </>
  )
}

InputButtonUI.propTypes = {
  data: PropTypes.string,
  getAppState: PropTypes.string,
  handleGetReportClick: PropTypes.func
}

export default InputButtonUI