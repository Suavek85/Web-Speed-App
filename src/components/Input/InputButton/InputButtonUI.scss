@import "../../../scss/styles.scss";

.inputbutton__wrapper {
    display: block;
    padding: 16px;
    margin-bottom: 20px;
    color: white;
    outline: none;
    border: none;
    &--inactive {
      background-color: rgba(55,71,133,.8);
      &:active {
        transform: translateY(2px);
        background-color:rgba(55,71,133,1)
      }
    }
    &--loading {
        background-color: $color-warning-loading;
    }
    &--error {
      background-color: $color-bottom-background;
    }
    &--success {
      background-color: $color-green-success;
    }
}