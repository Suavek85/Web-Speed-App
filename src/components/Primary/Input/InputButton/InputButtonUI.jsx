import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames/bind'

import { appStates } from '../../../../constants/states'
import { btnsContent } from '../../../../constants/content'
import styles from './InputButtonUI.scss'  

export default function InputButtonUI(props) {
 
  const { getAppState, handleGetReportClick } = props

  //BUTTON FORMATTING - CLASSES AND CONTENT
  const isStatus = state => getAppState === state
  
  const getContentFromStatus = state => { 
    const contentIndex = Object.keys(btnsContent).findIndex(el => el === state)
    return Object.values(btnsContent)[contentIndex]
  }

  let cx = classNames.bind(styles)

  let getButtonClasses = cx({
    block: true,
    blockInactive: isStatus(appStates.INACTIVE),
    blockLoading: isStatus(appStates.LOADING),
    blockError: isStatus(appStates.ERROR),
    blockSuccess: isStatus(appStates.SUCCESS),
  })

  return (
    <>
      <button 
        onClick={handleGetReportClick} 
        disabled={isStatus(appStates.LOADING)}
        className={getButtonClasses}
      >   
        {getContentFromStatus(getAppState)}
      </button>
    </>
  )
}

InputButtonUI.propTypes = {
  data: PropTypes.string,
  getAppState: PropTypes.string,
  handleGetReportClick: PropTypes.func
}
