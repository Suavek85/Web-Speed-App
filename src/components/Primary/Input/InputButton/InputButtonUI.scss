@import '../../../../scss/styles';

$block: inputbutton;
$blockInactive: modifier($block, inactive);
$blockLoading: modifier($block, loading);
$blockError: modifier($block, error);
$blockSuccess: modifier($block, success);
$blockMobile: modifier($block, mobile);

.#{$block} {
  display: block;
  width: 120px;
  height: 44px;
  margin-left: 50px;
  font-size: em(12);
  font-weight: 500;
  line-height: 44px;
  color: #fff;
  text-align: center;
  cursor: pointer;
  border: 0;
  outline: none;

  &--inactive {
    background-color: $color-blue;

    &:active {
      transform: translateY(2px);
    }
  }

  &--loading {
    background-color: $color-warning-loading;
  }

  &--error {
    width: 160px;
    background-color: $color-alert;
  }

  &--success {
    background-color: $color-green-success;
  }

  &--mobile {
    margin-top: 30px;
    margin-left: 0;
  }
}

:export {
  block: $block;
  blockInactive: $blockInactive;
  blockLoading: $blockLoading;
  blockError: $blockError;
  blockSuccess: $blockSuccess;
  blockMobile: $blockMobile;
}
