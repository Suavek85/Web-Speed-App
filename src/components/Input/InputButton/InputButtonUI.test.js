import React from 'react'
import { shallow } from 'enzyme'
import InputButtonUI from './InputButtonUI'

it('Should match the snapshot', () => {
  const wrapper = shallow(<InputButtonUI />).debug()
  expect(wrapper).toMatchSnapshot()
})