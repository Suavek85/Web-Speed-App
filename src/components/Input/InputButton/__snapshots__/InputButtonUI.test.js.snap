// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match the snapshot 1`] = `
"<Fragment>
  <button onClick={[undefined]} disabled={false} className=\\"inputbutton__wrapper\\" />
  <span />
</Fragment>"
`;
